src/main/scala/chisel3/util/HeterogeneousVec.scala

```diff
@@ -0,0 +1,121 @@
+// See LICENSE for license details.
+
+package chisel3.util
+
+import chisel3._
+import chisel3.core.{Data, requireIsChiselType, requireIsHardware}
+import chisel3.internal.naming.chiselName
+
+import scala.collection.immutable.ListMap
+
+object HeterogeneousVecWireInit {
+  /**
+    * Construct a new wire with the given bound values.
+    * This is analogous to [[chisel3.core.VecInit]].
+    * @param vals Values to create a HeterogeneousVec with and assign
+    * @return HeterogeneousVec with given values assigned
+    */
+  def apply[T <: Data](vals: Seq[T]): HeterogeneousVec[T] = {
+    // Create a wire of this type.
+    val hetVecWire = Wire(HeterogeneousVec(vals.map(_.cloneTypeFull)))
+    // Assign the given vals to this new wire.
+    for ((a, b) <- hetVecWire.zip(vals)) {
+      a := b
+    }
+    hetVecWire
+  }
+}
+
+object HeterogeneousVec {
+  /**
+    * Create a HeterogeneousVec from the type of the given Vec.
+    * For example, given a Vec(2, UInt(8.W)), this creates HeterogeneousVec(Seq.fill(2){UInt(8.W)}).
+    * @param vec Vec to use as template
+    * @return HeterogeneousVec analogous to the given Vec.
+    */
+  def apply[T <: Data](vec: Vec[T]): HeterogeneousVec[T] = {
+    HeterogeneousVec(Seq.fill(vec.length)(vec.sample_element))
+  }
+}
+
+/** Hardware module that is used to sequence n producers into 1 consumer.
+  * Priority is given to lower producer.
+  *
+  * @param eltsIn ...
+  *
+  * @example {{{
+  * val arb = Module(new Arbiter(UInt(), 2))
+  * arb.io.in(0) <> producer0.io.out
+  * arb.io.in(1) <> producer1.io.out
+  * consumer.io.in <> arb.io.out
+  * }}}
+  */
```

Comment:
I'm having trouble reconciling the previous comment with the following code.

Reply:
My bad - just wrote the actual scaladoc for the code.